Mix compile failed while importing busybox in nerves project

I am trying to use the library `vintage_net` which apparently requires busy box in a nerves project. I am using mac OS, elixir 1.9, otp 22. When I run `mix firmware` I get this error while compiling busybox:

```
==> busybox
mkdir -p /Users/adrien/Projects/insolo/mygadget/_build/rpi0_dev/lib/busybox/ebin/../obj
mkdir -p /Users/adrien/Projects/insolo/mygadget/_build/rpi0_dev/lib/busybox/ebin/../priv
curl -L https://www.busybox.net/downloads/busybox-1.30.1.tar.bz2 > /Users/adrien/Projects/insolo/mygadget/deps/busybox//busybox-1.30.1.tar.bz2
# Simple script for running "make menuconfig"
printf "#!/bin/sh\nKCONFIG_NOTIMESTAMP=1  CONFIG_PREFIX="/Users/adrien/Projects/insolo/mygadget/_build/rpi0_dev/lib/busybox/ebin/../priv" CROSS_COMPILE="/Users/adrien/.nerves/artifacts/nerves_toolchain_armv6_rpi_linux_gnueabi-darwin_x86_64-1.2.0/bin/armv6-rpi-linux-gnueabi-" SED=gsed -C /Users/adrien/Projects/insolo/mygadget/_build/rpi0_dev/lib/busybox/ebin/../obj menuconfig\ncp /Users/adrien/Projects/insolo/mygadget/_build/rpi0_dev/lib/busybox/ebin/../obj/.config /Users/adrien/Projects/insolo/mygadget/deps/busybox//busybox.config" > /Users/adrien/Projects/insolo/mygadget/deps/busybox//make_menuconfig
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 7611k  100 7611k    0     0   555k      0  0:00:13  0:00:13 --:--:-- 1371k
sha256sum -c busybox.hash
busybox-1.30.1.tar.bz2: OK
tar xf /Users/adrien/Projects/insolo/mygadget/deps/busybox//busybox-1.30.1.tar.bz2
touch /Users/adrien/Projects/insolo/mygadget/deps/busybox//busybox-1.30.1/.extracted
cd /Users/adrien/Projects/insolo/mygadget/deps/busybox//busybox-1.30.1; \
	for patchdir in /Users/adrien/Projects/insolo/mygadget/deps/busybox//patches/Darwin; do \
	    for patch in $(ls $patchdir); do \
		patch -p1 < "$patchdir/$patch"; \
	    done; \
	done
patching file scripts/gen_build_files.sh
KCONFIG_NOTIMESTAMP=1  -C /Users/adrien/Projects/insolo/mygadget/_build/rpi0_dev/lib/busybox/ebin/../obj KBUILD_SRC=/Users/adrien/Projects/insolo/mygadget/deps/busybox//busybox-1.30.1 CONFIG_PREFIX="/Users/adrien/Projects/insolo/mygadget/_build/rpi0_dev/lib/busybox/ebin/../priv" CROSS_COMPILE="/Users/adrien/.nerves/artifacts/nerves_toolchain_armv6_rpi_linux_gnueabi-darwin_x86_64-1.2.0/bin/armv6-rpi-linux-gnueabi-" SED=gsed -f /Users/adrien/Projects/insolo/mygadget/deps/busybox//busybox-1.30.1/Makefile defconfig
/bin/sh: -C: command not found
make: *** [/Users/adrien/Projects/insolo/mygadget/_build/rpi0_dev/lib/busybox/ebin/../obj/.config] Error 127
could not compile dependency :busybox, "mix compile" failed. You can recompile this dependency with "mix deps.compile busybox", update it with "mix deps.update busybox" or clean it with "mix deps.clean busybox"
```

Any idea on how to get over this? Thanks!

Thanks for your quick reply.

`brew install gmake`is not working but `brew install homebrew/core/make` is working to add `gmake`.

Now busybox is getting compiled but unfortunately it is getting stuck:

```
*
* Options common to all shells
*
*
* System Logging Utilities
*
klogd (5.7 kb) (KLOGD) [N/y/?] n
logger (6.3 kb) (LOGGER) [Y/n/?] y
logread (4.8 kb) (LOGREAD) [N/y/?] n
syslogd (13 kb) (SYSLOGD) [N/y/?] n
gmake[1]: Leaving directory '/Users/adrien/Projects/insolo/mygadget/_build/rpi0_dev/lib/busybox/obj'
KCONFIG_NOTIMESTAMP=1 /usr/local/bin/gmake CONFIG_PREFIX="/Users/adrien/Projects/insolo/mygadget/_build/rpi0_dev/lib/busybox/ebin/../priv" CROSS_COMPILE="/Users/adrien/.nerves/artifacts/nerves_toolchain_armv6_rpi_linux_gnueabi-darwin_x86_64-1.2.0/bin/armv6-rpi-linux-gnueabi-" SED=gsed -C /Users/adrien/Projects/insolo/mygadget/_build/rpi0_dev/lib/busybox/ebin/../obj
gmake[1]: Entering directory '/Users/adrien/Projects/insolo/mygadget/_build/rpi0_dev/lib/busybox/obj'
/usr/local/bin/gmake -C /Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1 O=/Users/adrien/Projects/insolo/mygadget/_build/rpi0_dev/lib/busybox/obj
  GEN     /Users/adrien/Projects/insolo/mygadget/_build/rpi0_dev/lib/busybox/obj/Makefile
  Using /Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1 as source for busybox
  SPLIT   include/autoconf.h -> include/config/*
  GEN     include/bbconfigopts.h
  GEN     include/common_bufsiz.h
  GEN     include/embedded_scripts.h
  HOSTCC  applets/usage
  HOSTCC  applets/applet_tables
  GEN     include/usage_compressed.h
  GEN     include/applet_tables.h include/NUM_APPLETS.h
  GEN     include/applet_tables.h include/NUM_APPLETS.h
  HOSTCC  applets/usage_pod
  CC      applets/applets.o
/Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1/applets/usage_pod.c:74:10: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
                printf(usage_array[i].aname);
                       ^~~~~~~~~~~~~~~~~~~~
/Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1/applets/usage_pod.c:74:10: note: treat the string as an argument to avoid this
                printf(usage_array[i].aname);
                       ^
                       "%s", 
1 warning generated.
  LD      applets/built-in.o
  DOC     busybox.pod
  DOC     BusyBox.txt
  DOC     busybox.1
  DOC     BusyBox.html
  LD      archival/built-in.o
  AR      archival/lib.a
  LD      archival/libarchive/built-in.o
  CC      archival/libarchive/common.o
  CC      archival/libarchive/data_align.o
  CC      archival/libarchive/data_extract_all.o
  CC      archival/libarchive/data_extract_to_stdout.o
  CC      archival/libarchive/data_skip.o
  CC      archival/libarchive/filter_accept_all.o
  CC      archival/libarchive/filter_accept_list.o
  LD      console-tools/built-in.o
  CC      archival/libarchive/filter_accept_reject_list.o
  CC      console-tools/chvt.o
  CC      archival/libarchive/find_list_entry.o
  CC      console-tools/clear.o
  CC      archival/libarchive/header_list.o
  CC      console-tools/deallocvt.o
  CC      archival/libarchive/header_skip.o
  CC      archival/libarchive/header_verbose_list.o
  CC      console-tools/dumpkmap.o
  CC      archival/libarchive/init_handle.o
  CC      console-tools/reset.o
  CC      console-tools/openvt.o
  CC      archival/libarchive/seek_by_jump.o
  CC      archival/libarchive/seek_by_read.o
  CC      console-tools/resize.o
  CC      console-tools/setconsole.o
  CC      archival/libarchive/unsafe_symlink_target.o
  CC      console-tools/setkeycodes.o
  LD      coreutils/built-in.o
  CC      coreutils/basename.o
  CC      coreutils/chgrp.o
  CC      coreutils/cat.o
  CC      coreutils/chmod.o
  CC      coreutils/chown.o
  CC      coreutils/chroot.o
  CC      coreutils/cksum.o
  CC      coreutils/cp.o
  LD      coreutils/libcoreutils/built-in.o
  CC      coreutils/cut.o
  CC      coreutils/libcoreutils/cp_mv_stat.o
  CC      coreutils/date.o
  CC      coreutils/libcoreutils/getopt_mk_fifo_nod.o
  CC      coreutils/dd.o
  CC      coreutils/df.o
  CC      coreutils/dirname.o
  CC      coreutils/du.o
  CC      coreutils/echo.o
  CC      coreutils/env.o
  CC      coreutils/expr.o
  CC      coreutils/factor.o
  CC      coreutils/false.o
  CC      coreutils/fold.o
  CC      coreutils/head.o
  CC      coreutils/hostid.o
  CC      coreutils/id.o
  CC      coreutils/install.o
  CC      coreutils/link.o
  CC      coreutils/ln.o
  CC      coreutils/logname.o
  CC      coreutils/ls.o
  LD      debianutils/built-in.o
  CC      debianutils/run_parts.o
  CC      debianutils/which.o
  LD      klibc-utils/built-in.o
  AR      klibc-utils/lib.a
  LD      e2fsprogs/built-in.o
  CC      e2fsprogs/chattr.o
  CC      e2fsprogs/e2fs_lib.o
  LD      editors/built-in.o
  CC      editors/awk.o
  CC      editors/cmp.o
  CC      editors/diff.o
  CC      editors/patch.o
  CC      editors/sed.o
  LD      findutils/built-in.o
  CC      findutils/find.o
  CC      findutils/grep.o
  CC      findutils/xargs.o
  LD      init/built-in.o
  CC      init/halt.o
  CC      e2fsprogs/fsck.o
  CC      e2fsprogs/lsattr.o
  LD      libbb/built-in.o
  CC      libbb/appletlib.o
  CC      libbb/ask_confirmation.o
  CC      libbb/auto_string.o
  CC      libbb/bb_askpass.o
  CC      libbb/bb_bswap_64.o
  CC      libbb/bb_cat.o
  CC      libbb/bb_do_delay.o
  CC      libbb/bb_getgroups.o
  CC      libbb/bb_getsockname.o
  CC      libbb/bb_pwd.o
  LD      libpwdgrp/built-in.o
  CC      libpwdgrp/uidgid_get.o
  CC      coreutils/md5_sha1_sum.o
  CC      coreutils/mkdir.o
  CC      coreutils/mkfifo.o
  CC      coreutils/mknod.o
  LD      loginutils/built-in.o
  AR      loginutils/lib.a
  LD      mailutils/built-in.o
  AR      mailutils/lib.a
  LD      miscutils/built-in.o
  CC      miscutils/bc.o
  CC      miscutils/crond.o
  CC      miscutils/crontab.o
  CC      miscutils/devmem.o
  CC      miscutils/partprobe.o
  CC      miscutils/setfattr.o
  CC      miscutils/setserial.o
  CC      miscutils/strings.o
  CC      miscutils/time.o
  CC      miscutils/watchdog.o
  CC      coreutils/mktemp.o
  CC      coreutils/mv.o
  CC      coreutils/nice.o
  CC      coreutils/nl.o
  CC      coreutils/nohup.o
  CC      coreutils/nproc.o
  LD      modutils/built-in.o
  AR      modutils/lib.a
  CC      coreutils/od.o
  CC      coreutils/paste.o
  CC      libbb/bb_qsort.o
  CC      libbb/bb_strtonum.o
  CC      libbb/capability.o
  CC      libbb/change_identity.o
  LD      networking/libiproute/built-in.o
  CC      networking/libiproute/ip_parse_common_args.o
  LD      networking/built-in.o
  CC      networking/arp.o
  CC      libbb/chomp.o
  CC      libbb/common_bufsiz.o
  CC      libbb/compare_string_array.o
  CC      coreutils/printenv.o
  CC      libbb/concat_path_file.o
  LD      networking/udhcp/built-in.o
  CC      networking/udhcp/dhcpd.o
  CC      networking/udhcp/dhcpc.o
  CC      networking/udhcp/arpping.o
  CC      networking/udhcp/common.o
  CC      coreutils/printf.o
  CC      coreutils/pwd.o
  CC      coreutils/readlink.o
gmake[4]: fork: Resource temporarily unavailable
  CC      coreutils/realpath.o
/bin/sh: fork: Resource temporarily unavailable
gmake[4]: *** [/Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1/scripts/Makefile.build:198: coreutils/realpath.o] Error 128
gmake[4]: *** Waiting for unfinished jobs....
gmake[4]: fork: Resource temporarily unavailable
  CC      networking/libiproute/ipaddress.o
/bin/sh: fork: Resource temporarily unavailable
  CC      networking/arping.o
/bin/sh: fork: Resource temporarily unavailable
gmake[4]: *** [/Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1/scripts/Makefile.build:198: networking/libiproute/ipaddress.o] Error 128
gmake[4]: *** Waiting for unfinished jobs....
gmake[4]: *** [/Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1/scripts/Makefile.build:198: networking/arping.o] Error 128
gmake[4]: *** Waiting for unfinished jobs....
  CC      networking/libiproute/iplink.o
/bin/sh: fork: Resource temporarily unavailable
/bin/sh: fork: Resource temporarily unavailable
gmake[3]: /Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1/Makefile:328: fork: Resource temporarily unavailable
gmake[3]: /Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1/Makefile:559: fork: Resource temporarily unavailable
gmake[3]: fork: Resource temporarily unavailable
gmake[4]: /Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1/scripts/Makefile.build:40: fork: Resource temporarily unavailable
gmake[4]: fork: Resource temporarily unavailable
gmake[3]: *** [/Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1/Makefile:743: procps] Error 2
gmake[3]: *** Waiting for unfinished jobs....
  CC      libbb/concat_subpath_file.o
gmake[4]: fork: Resource temporarily unavailable
  CC      networking/udhcp/domain_codec.o
/bin/sh: fork: Resource temporarily unavailable
gmake[4]: *** [/Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1/scripts/Makefile.build:198: networking/udhcp/domain_codec.o] Error 128
gmake[4]: *** Waiting for unfinished jobs....
  LD      printutils/built-in.o
  AR      printutils/lib.a
  AR      coreutils/libcoreutils/lib.a
  AR      console-tools/lib.a
  AR      archival/libarchive/lib.a
  AR      libpwdgrp/lib.a
  AR      init/lib.a
  AR      debianutils/lib.a
gmake[3]: *** [/Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1/Makefile:743: libbb] Error 2
gmake[3]: *** [/Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1/Makefile:743: networking] Error 2
gmake[3]: *** [/Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1/Makefile:743: networking/libiproute] Error 2
  AR      findutils/lib.a
  AR      e2fsprogs/lib.a
gmake[3]: *** [/Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1/Makefile:743: coreutils] Error 2
gmake[3]: *** [/Users/adrien/Projects/insolo/mygadget/deps/busybox/busybox-1.30.1/Makefile:743: networking/udhcp] Error 2
  AR      editors/lib.a
  AR      miscutils/lib.a
gmake[2]: *** [Makefile:112: _all] Error 2
gmake[1]: *** [Makefile:14: all] Error 2
gmake[1]: Leaving directory '/Users/adrien/Projects/insolo/mygadget/_build/rpi0_dev/lib/busybox/obj'
make: *** [install] Error 2
could not compile dependency :busybox, "mix compile" failed. You can recompile this dependency with "mix deps.compile busybox", update it with "mix deps.update busybox" or clean it with "mix deps.clean busybox"
==> mygadget
** (Mix) Could not compile with "make" (exit status: 2).
You need to have gcc and make installed. Try running the
commands "gcc --version" and / or "make --version". If these programs
are not installed, you will be prompted to install them.
```

I have tried to use `ulimit -u 2560` `ulimit -n 2560` to solve the compiling issue without success.

--- a/mix.exs
+++ b/mix.exs
@@ -11,7 +11,6 @@ defmodule Busybox.MixProject do
       compilers: [:elixir_make | Mix.compilers()],
       make_targets: ["all"],
       make_clean: ["clean"],
-      make_args: make_args(),
       make_error_message: "",
       start_permanent: Mix.env() == :prod,
       build_embedded: true,
@@ -63,10 +62,5 @@ defmodule Busybox.MixProject do
       source_ref: "v#{@version}",
       source_url: "https://github.com/nerves-networking/elixir_busybox"
     ]
-  end
-
-  defp make_args do
-    # Tell make to use all cores when building
-    ["-j", to_string(:erlang.system_info(:schedulers))]
   end
 end

